api/types/accesslist/review.go

```diff
@@ -89,6 +90,13 @@ func (r *Review) CheckAndSetDefaults() error {
 	r.SetKind(types.KindAccessListReview)
 	r.SetVersion(types.V1)
 
+	// If the name is not a UUID, generate a random UUID.
```

Comment:
Why do we need to support non-UUID, non-empty names? Can there be problems if a remote caller picks a specific UUID rather than us picking it?

Reply:
So realistically I don't see users picking their own names for this, but I came across a condition where, because the `service/local/access_list.go` was generating the UUIDs itself, the cache would never be identical, e.g. the same review would have two different UUIDs. So I need to maintain the name somehow -- counter to my previous thinking, I was thinking we just treat the names like normal (caller chooses the name) as opposed to what I was doing before. This bit is just a temporary hack until I can adjust the `e` tests not to work on the assumption that the name will always be replaced.

Reply:
As you mentioned over slack, it might be more appropriate to overwrite this name in the gRPC service instead of where I've got it. I still need this temporary hack so that I don't break the `e` tests, though.